src/security/tests/srv_acl_tests.c

```diff
@@ -1024,6 +1033,344 @@ test_check_pool_access_fall_thru_everyone(void **state)
+	daos_acl_free(acl);
+	daos_ace_free(ace);
+	daos_iov_free(&cred);
+}
+
+static void
+test_check_pool_access_user_beats_owner_grp(void **state)
+{
+	struct daos_acl		*acl;
+	struct daos_ace		*ace;
+	d_iov_t			cred;
+	struct pool_owner	ownership;
+
+	/* Owner group matches our creds */
+	ownership.user = "someuser@";
+	ownership.group = TEST_GROUP;
+
+	init_default_cred(&cred);
+
+	acl = get_acl_with_perms(DAOS_ACL_PERM_READ | DAOS_ACL_PERM_WRITE,
+				 DAOS_ACL_PERM_READ | DAOS_ACL_PERM_WRITE);
+
+	/* User entry matches our cred */
+	ace = daos_ace_create(DAOS_ACL_USER, TEST_USER);
+	ace->dae_access_types = DAOS_ACL_ACCESS_ALLOW;
+	ace->dae_allow_perms = DAOS_ACL_PERM_READ;
+	assert_int_equal(daos_acl_add_ace(&acl, ace), 0);
+
+	/*
+	 * Requesting RW - but user ACE has RO. User overrides owner-group
+	 * even though both match.
+	 * Owner-user doesn't match at all.
+	 */
+	assert_int_equal(ds_sec_check_pool_access(acl, &ownership, &cred,
+						  DAOS_PC_RW),
+			 -DER_NO_PERM);
+
+	daos_acl_free(acl);
+	daos_ace_free(ace);
+	daos_iov_free(&cred);
+}
+
+static void
+test_check_pool_access_grp_matches(void **state)
+{
+	struct daos_acl		*acl;
+	struct daos_ace		*ace;
+	d_iov_t			cred;
+	struct pool_owner	ownership;
+
+	/* Ownership won't match our creds */
+	ownership.user = "someuser@";
+	ownership.group = "somegroup@";
+
+	init_default_cred(&cred);
+
+	/* Group entry matches our cred */
+	ace = daos_ace_create(DAOS_ACL_GROUP, TEST_GROUP);
+	ace->dae_access_types = DAOS_ACL_ACCESS_ALLOW;
+	ace->dae_allow_perms = DAOS_ACL_PERM_READ | DAOS_ACL_PERM_WRITE;
+	acl = daos_acl_create(&ace, 1);
+
+	assert_int_equal(ds_sec_check_pool_access(acl, &ownership, &cred,
+						  DAOS_PC_RW),
+			 0);
+
+	daos_acl_free(acl);
+	daos_ace_free(ace);
+	daos_iov_free(&cred);
+}
+
+static void
+test_check_pool_access_grp_matches_second(void **state)
+{
+	struct daos_acl		*acl;
+	size_t			num_aces = 2;
+	struct daos_ace		*ace[num_aces];
+	d_iov_t			cred;
+	struct pool_owner	ownership;
+
+	/* Ownership won't match our creds */
+	ownership.user = "someuser@";
+	ownership.group = "somegroup@";
+
+	init_default_cred(&cred);
+
+	/* Match is not the first in the list */
+	ace[0] = daos_ace_create(DAOS_ACL_GROUP, "fakegrp@");
+	ace[0]->dae_access_types = DAOS_ACL_ACCESS_ALLOW;
+	ace[0]->dae_allow_perms = DAOS_ACL_PERM_READ;
+	ace[1] = daos_ace_create(DAOS_ACL_GROUP, TEST_GROUP);
+	ace[1]->dae_access_types = DAOS_ACL_ACCESS_ALLOW;
+	ace[1]->dae_allow_perms = DAOS_ACL_PERM_READ | DAOS_ACL_PERM_WRITE;
+	acl = daos_acl_create(ace, 2);
+
+	assert_int_equal(ds_sec_check_pool_access(acl, &ownership, &cred,
+						  DAOS_PC_RW),
+			 0);
+
+	daos_acl_free(acl);
+	free_ace_list(ace, num_aces);
+	daos_iov_free(&cred);
+}
+
+static void
+test_check_pool_access_grp_matches_multiple(void **state)
+{
+	struct daos_acl		*acl;
+	size_t			num_aces = 2;
+	struct daos_ace		*ace[num_aces];
+	d_iov_t			cred;
+	struct pool_owner	ownership;
+	static const char	*groups[] = { "group1@", "group2@" };
```

Comment:
(style)  static const char * array should probably be static const char * const

Reply:
This checkpatch suggestion seems a bit pedantic, and doesn't fit with our coding style. Requesting the gatekeeper ignore these ones.